### DISCUSSION TITLE
Comprehensive build system

### DISCUSSION THREAD

**Discussion by Drsheppard01** [AppImage/AppImageKit#1363] [Ideas]
A large number of projects use self-written build scripts as well as outdated tools such as [AppImageBuilder](https://github.com/AppImageCrafters/appimage-builder). Also some projects like vscodium use repackaging from deb format to AppImage using pkg2appimage, which does not support runtime, The current situation with new versions of glibc and fuse incompatibility is getting worse, forcing developers to abandon distribution in AppImage format.

See: https://github.com/prusa3d/PrusaSlicer/releases/tag/version_2.9.0-alpha1#Reason-for-moving-to-Flatpak
https://github.com/gaphor/gaphor/issues/2285

**Comment by probonopd** [comment]
Let's remember that AppImage is a self-mounting disk image that executes whatever the application developer has put inside it.

Within the AppImage project, we want to maintain the AppImageSpec and an implementation of the AppImage runtime. What goes inside an AppImage is out of the scope of the AppImage project and is best handled by the larger open source community.

As for your example, please test
https://github.com/probonopd/PrusaSlicer/releases

This utilizes https://github.com/VHSgunzo/sharun, which was conceived as an independent open source project.

**Reply by Drsheppard01** [reply]
When test this app i has this issue:

An error occured while setting up locale.
You may need to reconfigure the missing locales, likely by running the "locale-gen" and "dpkg-reconfigure locales" commands.
PrusaSlicer will now terminate.

locale::facet::_S_create_c_locale name not valid

**Reply by probonopd** [reply]
Thanks for testing. Which system are you on? And which language/locale is your system configured to use?

**Reply by Drsheppard01** [reply]
> Thanks for testing. Which system are you on? And which language/locale is your system configured to use?

Chimera Linux,

cat locale.conf
LANG='ru_RU.UTF-8'

LC_NUMERIC='ru_RU.UTF-8'
LC_TIME='ru_RU.UTF-8'
LC_MONETARY='ru_RU.UTF-8'
LC_PAPER='ru_RU.UTF-8'
LC_MEASUREMENT='ru_RU.UTF-8'

**Reply by probonopd** [reply]
Thanks. I did not know about Chimera Linux but I can reproduce the issue in NixOS.

![VirtualBox_NixOS 24 05 6668 e8c38b73aeb2 (x86_64-linux)_23_11_2024_23_20_47](https://github.com/user-attachments/assets/b340b34d-980f-4cc4-bb4a-656d0a9ca7b2)

Am on it.

**Reply by probonopd** [reply]
If I can have a guess, Chimera Linux does not have `/usr/lib/locale`?

**Reply by Drsheppard01** [reply]
If I'm not mistaken, you're talking about locales for glibc, while [Chimera Linux](https://chimera-linux.org/) doesn't use glibc and many parts of GNU

**Reply by probonopd** [reply]
Yes, I will have to figure out how to handle this.

**Reply by probonopd** [reply]
Bundling locales now, runs on Chimera Linux.

![image](https://github.com/user-attachments/assets/767b8f99-0d6d-4e6a-820b-9ae5c9dbf69a)

Chimera Linux is really interesting, I like it!
